registering matrix exponential breaks scalar exponential

```julia
julia> @register_array_symbolic Base.exp(x::AbstractMatrix) begin
           size=size(x)
       end

julia> @variables x
1-element Vector{Num}:
 x

julia> exp(x)
ERROR: AssertionError: (SymbolicUtils.symtype)(x) <: AbstractMatrix
Stacktrace:
 [1] exp
   @ ~/.julia/packages/Symbolics/TgrNY/src/wrapper-types.jl:202 [inlined]
 [2] exp(a::Num)
   @ Symbolics ~/.julia/packages/SymbolicUtils/UuDNI/src/methods.jl:71
 [3] top-level scope
   @ REPL[5]:1

```

Still an issue on latest SymbolicUtils
```julia
julia> @variables x[1:2, 1:2]
1-element Vector{Symbolics.Arr{Num, 2}}:
 x[1:2,1:2]

julia> exp(x)
ERROR: MethodError: no method matching exp!(::Matrix{Num})
The function `exp!` exists, but no method is defined for this combination of argument types.

Closest candidates are:
  exp!(::StridedMatrix{T}) where T<:Union{Float32, Float64, ComplexF64, ComplexF32}
   @ LinearAlgebra ~/.julia/juliaup/julia-1.12.4+0.x64.linux.gnu/share/julia/stdlib/v1.12/LinearAlgebra/src/dense.jl:726

Stacktrace:
 [1] exp(A::Symbolics.Arr{Num, 2})
   @ LinearAlgebra ~/.julia/juliaup/julia-1.12.4+0.x64.linux.gnu/share/julia/stdlib/v1.12/LinearAlgebra/src/dense.jl:671
 [2] top-level scope
   @ REPL[4]:1

julia> @register_array_symbolic Base.exp(x::AbstractMatrix) begin
                  size=size(x)
              end


julia> exp(x)
(exp(x))[1:2,1:2]

julia> @variables x
1-element Vector{Num}:
 x

julia> exp(x)
ERROR: AssertionError: (SymbolicUtils.symtype)(x) <: AbstractMatrix
Stacktrace:
 [1] exp
   @ ~/.julia/packages/Symbolics/2bB7u/src/wrapper-types.jl:202 [inlined]
 [2] exp(a::Num)
   @ Symbolics ~/.julia/packages/SymbolicUtils/kgV0N/src/methods.jl:71
 [3] top-level scope
   @ REPL[8]:1
```

--- a/src/array-lib.jl
+++ b/src/array-lib.jl
@@ -136,6 +136,9 @@ end
 
 Base.:(/)(x1::Num, x2::Arr{Num, 1}) = Arr{Num, 2}(unwrap(x1) / unwrap(x2))
 
+Base.exp(m::Matrix{Num}) = Arr{Num, 2}(exp(SConst(m)))
+Base.exp(m::Matrix{Complex{Num}}) = Arr{Complex{Num}, 2}(exp(SConst(m)))
+
 #################### MAP-REDUCE ################
 
 SymbolicUtils.@map_methods Arr unwrap wrap

--- a/test/arrays.jl
+++ b/test/arrays.jl
@@ -1,6 +1,6 @@
 using Symbolics
 using SymbolicUtils, Test
-using Symbolics: symtype, shape, wrap, unwrap, Arr, jacobian, @variables, value, get_variables, @arrayop, getname, metadata, scalarize
+using Symbolics: symtype, shape, wrap, unwrap, Arr, jacobian, @variables, value, get_variables, @arrayop, getname, metadata, scalarize, SConst
 using Base: Slice
 using SymbolicUtils: Sym, term, operation, search_variables
 import SymbolicUtils.Code: toexpr
@@ -533,4 +533,5 @@ end
 @testset "Issue#1751: `exp(::Arr)` works" begin
     @variables x[1:3, 1:3]
     @test isequal(exp(x), wrap(exp(unwrap(x))))
+    @test isequal(exp(collect(x)), exp(SConst(collect(unwrap(x)))))
 end

